tt_metal/impl/dispatch/kernels/command_queue_consumer.cpp

```diff
@@ -49,6 +49,7 @@ void kernel_main() {
         uint32_t consumer_cb_num_pages = command_ptr[DeviceCommand::consumer_cb_num_pages_idx];
         uint32_t num_pages = command_ptr[DeviceCommand::num_pages_idx];
         uint32_t producer_consumer_transfer_num_pages = command_ptr[DeviceCommand::producer_consumer_transfer_num_pages_idx];
+        bool is_sharded = (bool)command_ptr[DeviceCommand::buffer_type];
```

Comment:
weird way of doing things, and potentially error prone. I'm guessing this just works because sharded buffers map to 1, and interleaved to 0.

Reply:
cleaner way would be is_sharded = command_ptr[DeviceCommand::buffer_type] == (uint32_t)BufferType::SHARDED